ledger/tests/helpers/mod.rs

```diff
@@ -0,0 +1,290 @@
+// Copyright (c) 2019-2025 Provable Inc.
+// This file is part of the snarkVM library.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use aleo_std::StorageMode;
+use console::{
+    account::{Address, PrivateKey},
+    network::MainnetV0,
+    prelude::*,
+};
+use ledger_narwhal::{BatchCertificate, BatchHeader, Subdag};
+use ledger_store::ConsensusStore;
+use snarkvm_ledger::{Block, Ledger};
+use synthesizer::vm::VM;
+use time::OffsetDateTime;
+
+use indexmap::{IndexMap, IndexSet};
+use std::collections::{BTreeMap, HashMap};
+
+pub type CurrentNetwork = MainnetV0;
+
+#[cfg(not(feature = "rocks"))]
+pub type LedgerType<N> = ledger_store::helpers::memory::ConsensusMemory<N>;
+#[cfg(feature = "rocks")]
+pub type LedgerType<N> = ledger_store::helpers::rocksdb::ConsensusDB<N>;
+
+/// Helper to build chains with custom structures for testing
+pub struct TestChainBuilder {
+    /// The keys of all validators.
+    private_keys: Vec<PrivateKey<CurrentNetwork>>,
+    /// The underlying ledger.
+    ledger: Ledger<CurrentNetwork, LedgerType<CurrentNetwork>>,
+    /// The round containing the leader certificate for the most recent block we generated.
+    last_block_round: u64,
+    /// The batch certificates of the last round we generated.
+    round_to_certificates: HashMap<u64, IndexMap<usize, BatchCertificate<CurrentNetwork>>>,
+    /// The batch certificate of the last leader (if any).
+    previous_leader_certificate: Option<BatchCertificate<CurrentNetwork>>,
+    /// The height of the last block we created.
+    current_height: u64,
+    /// The last batch for each committee member that was included in a block.
+    /// Maps the author's index to a round number.
+    last_batch: HashMap<usize, u64>,
+    /// The last batch of a validator that was included in a block
+    last_committed_batch: HashMap<usize, u64>,
```

Comment:
What's the exact difference between `last_batch` and `last_committed_batch`?

Reply:
The doc is incorrect and I will fix it. last_committed_batch is for batches included in a block, and last_batch for any certificate created by the validator.